flocker/node/test/test_script.py

```diff
@@ -254,23 +291,135 @@ def test_custom_port(self):
             The ``--destination-port`` command-line option allows configuring
             the destination port.
             """
-            self.options.parseOptions([b"--destination-port", b"1234",
-                                       b"127.0.0.1"])
+            self.options.parseOptions([b"--destination-port", b"1234"])
             self.assertEqual(self.options["destination-port"], 1234)
 
-        def test_host(self):
+        def test_default_config_file(self):
             """
-            The second required command-line argument allows configuring the
-            destination host.
+            The default config file is a FilePath with path
+            ``/etc/flocker/agent.yml``.
             """
-            self.options.parseOptions([b"10.0.0.1"])
+            self.options.parseOptions([])
             self.assertEqual(
-                self.options["destination-host"], u"10.0.0.1",
-            )
+                self.options["control-service-config"],
+                FilePath("/etc/flocker/agent.yml"))
+
+        def test_custom_config_file(self):
+            """
+            The ``--config-file`` command-line option allows configuring
+            the config file.
+            """
+            self.options.parseOptions(
+                [b"--control-service-config", b"/etc/foo.yml"])
+            self.assertEqual(
+                self.options["control-service-config"],
+                FilePath("/etc/foo.yml"))
 
     return Tests
 
 
+class ConfigurationFromOptionsTests(SynchronousTestCase):
+    """
+    Tests for :func:`configuration_from_options`.
+    """
+
+    def setUp(self):
+        self.scratch_directory = FilePath(self.mktemp())
+        self.scratch_directory.makedirs()
+        self.config = self.scratch_directory.child('config.yml')
+        self.options = {
+            'control-service-config': self.config,
+        }
+
+    def test_error_on_file_does_not_exist(self):
```

Comment:
All the error tests are very repetitive. Add a new `assertErrorForYAML` utility method, at least.